pkg/zonemgr/zone_file_cache_test.go

```diff
@@ -38,6 +41,20 @@ var _ = Describe("cached zone file content maintenance", func() {
 			Entry("header should contain default values", "", "", headerDefault),
 			Entry("header should contain custom values", nameServerIP, domain, headerCustom),
 		)
+
+		When("zone file already exist", func() {
+			BeforeEach(func() {
+				Expect(os.Mkdir("zones", 0777)).To(Succeed())
+				Expect(os.WriteFile(zoneFileName, []byte(content), 0777)).To(Succeed())
+			})
+			AfterEach(func() {
+				Expect(os.RemoveAll("zones")).To(Succeed())
+			})
+
+			It("should set SOA serial equal to existing value", func() {
```

Comment:
I think the method the has to be tested is the prive one `NewZoneFileCache` and not an internal method.
I would call `NewZoneFileCache`and check the `header`. Same as the other tests.

Reply:
Doesn't seem this comment was handled.

Reply:
fixed